engine/src/main/java/org/hibernate/search/query/engine/impl/QueryFilterset.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.query.engine.impl;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.BooleanClause.Occur;
+
+/**
+ * Internal representation for a set of filters to be applied on a query.
+ *
+ * @author Sanne Grinovero
+ */
+public final class QueryFilterset {
+
+	public static final QueryFilterset EMPTY_FILTERSET = new QueryFilterset();
+
+	private final List<Query> filterQueries;
+
+	private QueryFilterset() {
+		filterQueries = Collections.emptyList();
+	}
+
+	public QueryFilterset(List<Query> filterQueries) {
+		this.filterQueries = filterQueries;
+	}
+
+	/**
+	 * Will wrap the passed Query into a BooleanQuery to apply all filters as boolean clauses.
+	 * Returns the unmodified (same instance) of the input if there are no filters to be applied.
+	 * @param filteredQuery
+	 * @return
+	 */
+	public Query filterOrPassthrough(Query filteredQuery) {
+		if ( isEmpty() ) {
+			return filteredQuery;
+		}
+		else {
+			BooleanQuery.Builder boolQueryBuilder = new BooleanQuery.Builder();
+			boolQueryBuilder.add( filteredQuery, Occur.MUST );
+			for ( Query bc : filterQueries ) {
+				boolQueryBuilder.add( bc, BooleanClause.Occur.FILTER );
+			}
+			return boolQueryBuilder.build();
+		}
+	}
+
+	/**
+	 * @return lists all Lucene query instances which should be applied as filters
+	 */
+	public List<Query> filterQueries() {
```

Comment:
I guess that's because of a previous implementation: it didn't look like a getter at all.
This was more complex than just exposing the reference to a pre-built list.

Should I change it?